### PR TITLE
Affinity fix version C

### PR DESCRIPTION
**Pull request by fireice-uk**
Order of calls in this PR:

1. bindMemoryToNUMANode
2. thd_setaffinity, not in this thread

### CODE REVIEW

**Comment by psychocrypt** [comment]
All three versions are working without any issues. I prefer version B, @fireice-uk what is your favorite patch?
The version B is easy to read and more intuitive than version C with the feature.

**Comment by fireice-uk** [comment]
Looks like for Windows to be happy the order of calls needs to be like in this patch. 

I find it puzzling however that all of them would work. Can you give this one a bit more testing on the systems that were affected on your side?

**Comment by psychocrypt** [comment]
Yes I will look deeper to this PR, but could be not before end of the next
week.

File: xmrstak/backend/cpu/minethd.cpp [hunk 1]
```diff
@@ -317,23 +308,17 @@ minethd::cn_hash_fun minethd::func_selector(bool bHaveAes, bool bNoPrefetch)
...
-#if defined(__APPLE__)
-	printer::inst()->print_msg(L1, "WARNING on MacOS thread affinity is only advisory.");
-#endif
-	thd_setaffinity(oWorkThd.native_handle(), affinity);
 }
 
 void minethd::work_main()
 {
 	if(affinity >= 0) //-1 means no affinity
 		pin_thd_affinity();
```
Comment (psychocrypt) [inline xmrstak/backend/cpu/minethd.cpp]: `bindMemoryToNUMANode(affinity);` should be called here directly and `pin_thd_affinity` can be removed.

Comment (fireice-uk) [reply xmrstak/backend/cpu/minethd.cpp]: Ok. I will test that one on Windows too.

**Comment by fireice-uk** [comment]
I think this one is done. Let me know.

**Pull request merged**